File: src/types/events.ts
export interface SessionOutputEvent {
  event: string;
  payload: SessionOutputPayload;
}

export interface SessionOutputPayload {
  session_id: string;
  kind: SessionOutputKind;
}

export type SessionOutputKind =
  | { type: "stdout"; data: string }
  | { type: "exited"; data: number };

File: src-tauri/src/session/events.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SessionOutputEvent {
    pub session_id: String,
    pub kind: SessionOutputKind,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(tag = "type", content = "data")]
pub enum SessionOutputKind {
    stdout(String),
    exited(u32),
}
